fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java

```diff
@@ -1600,28 +1601,41 @@ public AlterJobV2 createAlterMetaJob(List<AlterClause> alterClauses, Database db
                 throw new DdlException("Only support alter one property in one stmt");
             }
 
-            boolean enablePersistentIndex = false;
             if (properties.containsKey(PropertyAnalyzer.PROPERTIES_ENABLE_PERSISTENT_INDEX)) {
-                enablePersistentIndex = PropertyAnalyzer.analyzeBooleanProp(properties,
+                boolean enablePersistentIndex = PropertyAnalyzer.analyzeBooleanProp(properties,
                         PropertyAnalyzer.PROPERTIES_ENABLE_PERSISTENT_INDEX, false);
                 boolean oldEnablePersistentIndex = olapTable.enablePersistentIndex();
                 if (oldEnablePersistentIndex == enablePersistentIndex) {
                     LOG.info(String.format("table: %s enable_persistent_index is %s, nothing need to do",
                             olapTable.getName(), enablePersistentIndex));
                     return null;
                 }
+
+                long timeoutSecond = PropertyAnalyzer.analyzeTimeout(properties, Config.alter_table_timeout_second);
+                alterMetaJob = new LakeTableAlterMetaJob(GlobalStateMgr.getCurrentState().getNextId(),
+                        db.getId(),
+                        olapTable.getId(), olapTable.getName(), timeoutSecond,
+                        TTabletMetaType.ENABLE_PERSISTENT_INDEX, enablePersistentIndex);
+            } else if (properties.containsKey(PropertyAnalyzer.PROPERTIES_DATACACHE_PARTITION_DURATION)) {
+                PeriodDuration partitionDuration = PropertyAnalyzer.analyzeDataCachePartitionDuration(properties);
+                if (partitionDuration == null) {
+                    throw new DdlException("Null datacache.partition_duration");
+                }
+
+                PeriodDuration oldPartitionDuration = olapTable.dataCachePartitionDuration();
+                if (partitionDuration.equals(oldPartitionDuration)) {
+                    LOG.info(String.format("table: %s datacache.partition_duration is %s, nothing need to do",
+                            olapTable.getName(), partitionDuration));
+                    return null;
+                }
+                olapTable.setDataCachePartitionDuration(partitionDuration);
```

Comment:
1. need write edit log after update.
2. partition duration is also need to update.
`ALTER TABLE table modify partition (p1) set ("xxx"="yyy")`